prover/prover.go

```diff
@@ -232,7 +233,8 @@ func sequentialWork(
 
 	for {
 		// Generate the next leaf.
-		err := tree.AddLeaf(makeLabel(labelHashFunc, nextLeafID, tree.GetParkedNodes()))
+		parkedNodes = tree.GetParkedNodes(parkedNodes[:0])
```

Comment:
again, I'm no 100% sure about this but `parkedNodes[:0]` is a slice with (in this case) len 0 and cap 0, `nil` should have the same effect.

Reply:
A `parkedNodes[:0]` is a slice with length 0 but capacity equal to capacity of `parkedNodes`. The idea is that `parkedNodes` local is reused in each loop iteration avoiding reallocating memory every time (the result is reassigned back to `parkedNodes` exactly like `append()` works.

Reply:
I see, can the capacity be estimated before calling this function? On the first call `parkedNodes` is `nil` and the slice is created here with `cap` 0 and `len` 0 (to then be re-allocated by the `tree.GetParkedNodes` function.

If `tree.GetParkedNodes` allocates enough capacity this will only happen here and is only 1 allocation that could be saved on the first iteration of the loop.

Reply:
Yes, you are right - it's `nil` on the first iteration. Then it is resized (re-allocated) if required in `tree.GetParkedNodes()`. It is, however, difficult to predict the required size for the slice. It grows with the tree but rather slowly. IIRC, it's 1 parked node per tree layer which is log2(tree width) - negligible (according to my quick calculation it will be max 38 elements for 300k leaves/sec and 2w epoch).